contracts/HypERC721Collateral.sol

```diff
@@ -10,10 +10,10 @@ import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
  * @author Abacus Works
  */
 contract HypERC721Collateral is TokenRouter {
-    IERC721 public immutable wrappedToken;
+    address public immutable wrappedToken;
```

Comment:
Why change this to `address`?

Reply:
cant cast from IERC721 to IERC721Metadata